files/walle.py

```diff
@@ -404,17 +428,23 @@ def load_malware_lib_from_env(malware_file: pathlib.Path) -> dict:
 
 def digest_file_crc32(chunksize: int, path: pathlib.Path) -> int:
     crc32 = 0
-    with open(path, "rb") as specimen:
-        while chunk := specimen.read(chunksize):
-            crc32 = zlib.crc32(chunk, crc32)
+    try:
+        with open(path, "rb") as specimen:
+            while chunk := specimen.read(chunksize):
+                crc32 = zlib.crc32(chunk, crc32)
+    except PermissionError:
+        logger.warning(f"Permission denied for file: {path}")
     return crc32
 
 
 def digest_file_sha1(chunksize: int, path: pathlib.Path) -> str:
     sha1 = hashlib.sha1()
-    with open(path, "rb") as specimen:
-        while chunk := specimen.read(chunksize):
-            sha1.update(chunk)
+    try:
+        with open(path, "rb") as specimen:
+            while chunk := specimen.read(chunksize):
+                sha1.update(chunk)
+    except PermissionError:
+        logger.warning(f"Permission denied for file: {path}")
```

Comment:
There are two cases where I caught permission errors here, in my case it was only for one file in the JWD (I think command.sh) but this error could be fatal if all JWD files are PermissionDenied. I guess in that case it would be pretty obvious in walle.log that walle is not working.

Reply:
Yep, in my `walle.log` I get:

```
2024-10-11 04:48 - WARNING - Permission denied for file: /mnt/galaxy/tmp/job_working_directory/_interactive/24546/command.sh
```

Reply:
Thank you! I think walle should run as root, because the jupyter users have root access inside their jupyter notebook and can save files as uid and gid 0

Reply:
WallE could cleanup everything non-root and the rest we could leave to our normal cleanup scripts?